sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -262,11 +262,15 @@ private[hive] class SparkExecuteStatementOperation(
       // HiveServer will silently swallow them.
       case e: Throwable =>
         val currentState = getStatus().getState()
-        logError(s"Error executing query, currentState $currentState, ", e)
-        setState(OperationState.ERROR)
-        HiveThriftServer2.listener.onStatementError(
-          statementId, e.getMessage, SparkUtils.exceptionString(e))
-        throw new HiveSQLException(e.toString)
+        if (currentState == OperationState.CANCELED) {
+          return
```

Comment:
A client can call cancel() and close() before the execution thread reaches here.
Could you do:
```
        val currentState = getStatus().getState()
        if ((currentState == OperationState.CANCELED)
            || (currentStateg == OperationState.CLOSED)
            || (currentState == OperationState.FINISHED)) {
          // This may happen if the execution was cancelled, and then closed from another thread.
          logWarning(s"Ignore exception in terminal state with $statementId: $e")
          return
        }
```
This follows https://github.com/apache/hive/blob/master/service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java#L247

Reply:
resonable, run statement async , this situation can truly happen.